plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammar310/python.jjt_template

```diff
@@ -920,198 +920,167 @@ void nonlocal_stmt(): {}
+void closed_pattern() #void : {}
 {
-    <LBRACKET> [maybe_sequence_pattern()] <RBRACKET>
-    | <LPAREN> [open_sequence_pattern()] <RPAREN>
+    literal_pattern()
+    | class_pattern()
+    | sequence_pattern()
+    | mapping_pattern()
+    | wildcard_pattern()
 }
 
-void group_pattern() #void : {}
+void literal_pattern() #void : {}
 {
-    <LPAREN> pattern() <RPAREN>
+    signed_number() (
+        <PLUS> signed_number()
+        | <MINUS> signed_number()
+    )?
+    | String()
+    | <NONE>
+    | <TRUE>
+    | <FALSE>
 }
 
-void attr() #void : {}
+void signed_number() #void : {}
 {
-    Name() (<DOT> Name())*
+    Number()
+    | <MINUS> Number()
 }
 
-void value_pattern() #void : {}
+void class_pattern() : {}
 {
-    attr()
+    attr() [<LPAREN> [pattern_arguments() [<COMMA>]] <RPAREN>]
 }
 
-void wildcard_pattern() #void : {}
+void attr() : {}
 {
-    "_"
+    Name() (
+        <EQUAL> pattern()
+        | (<DOT> Name())*
+    )
 }
 
-void capture_pattern() #void : {}
+void pattern_arguments() #void : {}
 {
-    Name()
+    pattern() (<COMMA> pattern())*
 }
 
-void signed_number() #void : {}
+void sequence_pattern() : {}
 {
-    Number()
-    | <MINUS> Number()
+    list_pattern()
+    | tuple_pattern()
 }
 
-void literal_pattern() #void : {}
+void list_pattern() #void : {}
 {
-    LOOKAHEAD(signed_number() <PLUS>) signed_number() <PLUS> Number()
-    | LOOKAHEAD(signed_number() <MINUS>) signed_number() <MINUS> Number()
-    | LOOKAHEAD(signed_number()) signed_number()
-    | String()
-    | <NONE>
-    | <TRUE>
-    | <FALSE>
+    <LBRACKET> [open_sequence_pattern()] <RBRACKET>
 }
 
-void closed_pattern() #void : {}
+void tuple_pattern() #void : {}
 {
-    LOOKAHEAD(literal_pattern()) literal_pattern()
-    | LOOKAHEAD(class_pattern()) class_pattern()
-    | LOOKAHEAD(group_pattern()) group_pattern()
-    | LOOKAHEAD(sequence_pattern()) sequence_pattern()
-    | LOOKAHEAD(mapping_pattern()) mapping_pattern()
-    | LOOKAHEAD(value_pattern()) value_pattern()
-    | LOOKAHEAD(capture_pattern()) capture_pattern()
-    | LOOKAHEAD(wildcard_pattern()) wildcard_pattern()
+    <LPAREN> (
+        [open_sequence_pattern()]
+        | pattern()
+    ) <RPAREN>
 }
 
-void or_pattern() #void : {}
+void mapping_pattern() : {}
 {
-    closed_pattern() (<OR> closed_pattern())*
+    <LBRACE> [items_pattern()] <RBRACE>
 }
 
-void as_pattern() #void : {}
+void items_pattern() #void : {}
 {
-    or_pattern() $AS capture_pattern()
+    key_value_pattern() (<COMMA> key_value_pattern())* [<COMMA>]
 }
 
-void pattern() #void : {}
+void key_value_pattern() : {}
 {
-    LOOKAHEAD(as_pattern()) as_pattern()
-    | or_pattern()
+    (
+        literal_pattern()
+        | attr()
+    ) <COLON> pattern()
+    | double_star_pattern()
 }
 
-void patterns() #void : {}
+void double_star_pattern() #void : {}
 {
-    LOOKAHEAD(open_sequence_pattern()) open_sequence_pattern()
-    | pattern()
+    <POWER> Name()
 }
 
-void guard() #void : {}
+void guard() : {}
```

Comment:
Is there any reason for making the `guard` a rule returning an AST node? I think it could be just `#void` as it was...